packages/api/src/rx/SubmittableExtrinsic.ts

```diff
@@ -25,15 +22,34 @@ export default class SubmittableExtrinsic extends Extrinsic {
     this._api = api;
   }
 
-  send (): Observable<SendResult> {
+  // FIXME split into graph derivation once available
+  private trackStatus = (status: ExtrinsicStatus): Observable<SubmittableSendResult> => {
+    if (status.type !== 'Finalised') {
+      return of({
+        status,
+        type: status.type
+      });
+    }
+
+    const blockHash = status.value as Hash;
+
+    return combineLatest(
+      this._api.rpc.chain.getBlock(blockHash),
+      this._api.query.system.events.at(blockHash)
+    ).pipe(
+      // @ts-ignore just wtf?
```

Comment:
not 100% either, but this works:

```
    return combineLatest(
      this._api.rpc.chain.getBlock(blockHash) as Observable<SignedBlock>,
      this._api.query.system.events.at(blockHash) as Observable<Vector<EventRecord>>
    ).pipe(
      map(([signedBlock, allEvents]) => ({
        events: filterEvents(this.hash, signedBlock, allEvents),
        status,
        type: status.type
      }))
    );
```

ideally getBlock should really return a `Observable<SignedBlock>`, instead of `RpcRxInterface$Method`